cdc/scheduler/internal/v3/agent/agent.go

```diff
@@ -201,26 +201,26 @@ func NewAgent(ctx context.Context,
 }
 
 // Tick implement agent interface
-func (a *agent) Tick(ctx context.Context) error {
+func (a *agent) Tick(ctx context.Context) (*schedulepb.Barrier, error) {
 	inboundMessages, err := a.recvMsgs(ctx)
 	if err != nil {
-		return errors.Trace(err)
+		return nil, errors.Trace(err)
 	}
 
-	outboundMessages := a.handleMessage(inboundMessages)
+	outboundMessages, barrier := a.handleMessage(inboundMessages)
 
 	responses, err := a.tableM.poll(ctx)
 	if err != nil {
-		return errors.Trace(err)
+		return nil, errors.Trace(err)
 	}
 
 	outboundMessages = append(outboundMessages, responses...)
 
 	if err := a.sendMsgs(ctx, outboundMessages); err != nil {
-		return errors.Trace(err)
+		return nil, errors.Trace(err)
 	}
 
-	return nil
+	return barrier, nil
```

Comment:
What happens when barrier falls back?

Reply:
`barrier` is used by `sinkManager` to update tableSink's `barrierTs`, it will be check if it's laeger than the last `barrierTs` before update.

Reply:
This logic will be added in next PR.